src/main/kotlin/de/timo_reymann/ansible_vault_integration/intention/menu/VaultFileMenuAction.kt

```diff
@@ -1,70 +1,84 @@
 package de.timo_reymann.ansible_vault_integration.intention.menu
 
+import com.intellij.notification.Notification
+import com.intellij.notification.NotificationType
+import com.intellij.notification.Notifications
 import com.intellij.openapi.actionSystem.AnAction
 import com.intellij.openapi.actionSystem.AnActionEvent
 import com.intellij.openapi.actionSystem.LangDataKeys
 import com.intellij.openapi.progress.ProgressManager
+import com.intellij.psi.PsiBinaryFile
 import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiManager
 import de.timo_reymann.ansible_vault_integration.config.AnsibleConfigurationService
+import de.timo_reymann.ansible_vault_integration.config.VaultIdentity
 import de.timo_reymann.ansible_vault_integration.execution.AnsibleVaultTask
 import de.timo_reymann.ansible_vault_integration.intention.AnsibleVaultIdentityPopup
-import de.timo_reymann.ansible_vault_integration.runnable.AnsibleVaultRunnable
 import de.timo_reymann.ansible_vault_integration.runnable.DecryptFileAnsibleVaultRunnable
 import de.timo_reymann.ansible_vault_integration.runnable.EncryptFileAnsibleVaultRunnable
-import org.jetbrains.yaml.psi.YAMLFile
-import setVisible
 
 open class VaultFileMenuAction : AnAction() {
+    private val progressManager: ProgressManager = ProgressManager.getInstance()
+
     override fun actionPerformed(e: AnActionEvent) {
-        val psiFile = e.getData(LangDataKeys.PSI_FILE) ?: return
-        val content = psiFile.text ?: return
-        val isVaulted = content.startsWith("\$ANSIBLE_VAULT")
-        val title = "${if (isVaulted) "Decrypt" else "Encrypt"} file with ansible-vault"
+        val project = e.project ?: return
+        val psiManager = PsiManager.getInstance(project)
+        val vaultIdentities = AnsibleConfigurationService.getInstance(project).getAggregatedConfig().vaultIdentities
 
-        if (isVaulted) {
-            runTask(psiFile, title, DecryptFileAnsibleVaultRunnable(psiFile.project, psiFile, content))
-        } else {
-            val vaultIdentities = AnsibleConfigurationService.getInstance(psiFile.project)
-                .getAggregatedConfig()
-                .vaultIdentities
+        e.getData(LangDataKeys.VIRTUAL_FILE_ARRAY)?.forEach {
+            val psiFile = psiManager.findFile(it) ?: return@forEach
 
-            if (vaultIdentities != null && vaultIdentities.isNotEmpty()) {
-                AnsibleVaultIdentityPopup(vaultIdentities) {
-                    runTask(
-                        psiFile,
-                        title,
-                        EncryptFileAnsibleVaultRunnable(psiFile.project, psiFile, content, it, false)
-                    )
-                }.showCentered()
-            } else {
-                runTask(
-                    psiFile,
-                    title,
-                    EncryptFileAnsibleVaultRunnable(psiFile.project, psiFile, content, null, false)
-                )
+            if (psiFile is PsiBinaryFile) {
```

Comment:
chore: Move the check in `#update` method

I would go with not supporting binary files first, so the mutli-file support is there, as you already suggested.

Simply hiding the context action when a binary file is selected would be more the IntelliJ way. Find it cleaner to have an action not available when it's well known it can't fully be performed.